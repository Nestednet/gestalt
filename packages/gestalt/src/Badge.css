.badge {
  padding: 6px 10px;
}

.pill {
  border-radius: 500px;
}

.rounded-rectangle {
  border-radius: 4px;
}

.warning {
  background: #f9ecc9;
  color: #62321f;
}

.alarm {
  background: #fee0e0;
  color: #6e2425;
}

.good {
  background: #d5ede9;
  color: #138771;
}

.info {
  background: #e8e8e8;
  color: #45575b;
}

.center {
  text-align: center;
}

.left {
  text-align: left;
}

.right {
  text-align: right;
}

.justify {
  text-align: justify;
}
