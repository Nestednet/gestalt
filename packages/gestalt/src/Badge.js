// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Badge.css';
import Text from './Text.js';

type Props = {|
  align?: 'left' | 'right' | 'center' | 'justify',
  children?: React.Node,
  status?: 'warning' | 'alarm' | 'good' | 'info',
  inline?: boolean,
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  shape?: 'pill' | 'rounded-rectangle',
|};

export default function Badge({
  align = 'center',
  status = 'info',
  shape = 'pill',
  children,
  inline = true,
  size = 'md',
}: Props) {
  const classes = classnames(
    styles.badge,
    styles[shape],
    styles[status],
    styles[align]
  );

  const Tag = inline ? 'span' : 'div';
  return (
    <Tag className={classes}>
      <Text size={size}>{children}</Text>
    </Tag>
  );
}

Badge.propTypes = {
  align: PropTypes.oneOf(['left', 'right', 'center', 'justify']),
  status: PropTypes.oneOf(['warning', 'alarm', 'good', 'info']),
  inline: PropTypes.bool,
  children: PropTypes.node,
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
};
