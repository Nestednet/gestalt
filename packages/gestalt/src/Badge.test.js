// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Badge from './Badge.js';

it('Badge renders', () => {
  const component = renderer.create(<Badge>Test</Badge>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
