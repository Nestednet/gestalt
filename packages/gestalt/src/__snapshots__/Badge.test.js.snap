// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders 1`] = `
<span
  className="badge pill info center"
>
  <div
    className="Text fontSize3 darkGray alignLeft breakWord fontStyleNormal fontWeightNormal"
  >
    Test
  </div>
</span>
`;
